src/Microsoft.Graph.Core/Requests/HttpProvider.cs

```diff
@@ -203,8 +203,16 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request)
                             error.ThrowSite = throwsiteValues.FirstOrDefault();
                         }
                     }
+                    
+                    throw new ServiceException(error)
+                    {
+                        // Pass through the response headers to the ServiceException.
+                        ResponseHeaders = response.Headers,
 
-                    throw new ServiceException(error);
+                        // System.Net.HttpStatusCode does not support RFC 6585, Additional HTTP Status Codes.
+                        // Throttling status code 429 is in RFC 6586. The status code 429 will be passed through.
```

Comment:
I would clean up this comment by removing the reference to the 429 error code, since it affects other status codes that are not defined by the enum